LoopKit/InsulinKit/DoseStore.swift

```diff
@@ -831,6 +835,107 @@ extension DoseStore {
         }
     }
 
+    /**
+     Adds and persists doses
+     - parameter doses: An array of dose entries to add.
+     - parameter completion: A closure called after the doses are saved. The closure takes a single argument:
+     - parameter error: An error object explaining why the doses could not be saved.
+     */
+    public func addDoses(_ doses: [DoseEntry], completion: @escaping (_ error: Error?) -> Void) {
+        guard doses.count > 0 else {
+            completion(nil)
+            return
+        }
+
+        self.persistenceController.save { (error) -> Void in
+            self.insulinDeliveryStore.addDoseEntries(doses, from: nil, syncVersion: self.syncVersion) { (result) in
+                switch result {
+                case .success:
+                    completion(nil)
+                    self.syncPumpEventsToInsulinDeliveryStore { error in
+                        completion(error)
+                        NotificationCenter.default.post(name: DoseStore.valuesDidChange, object: self)
+                    }
+                case .failure(let error):
+                    self.log.error("Error adding dose: %{public}@", String(describing: error))
+                    completion(error)
+                }
+            }
+        }
+    }
+
+    /// Deletes one particular manually entered dose from the store
+    ///
+    /// - Parameter dose: Dose to delete.
+    /// - Parameter completion: A closure called after the event deleted. This closure takes a single argument:
+    /// - Parameter success: True if dose was successfully deleted
+    public func deleteDose(_ dose: DoseEntry, completion: @escaping (_ error: DoseStoreError?) -> Void) {
+        guard let syncIdentifier = dose.syncIdentifier else {
+            self.log.error("Unable to delete PersistedManualEntryDose: no syncIdentifier")
+            completion(DoseStoreError.fetchError(description: "Unable to delete dose: syncIdentifier is missing", recoverySuggestion: "File an issue report in Github"))
+            return
+        }
+        insulinDeliveryStore.deleteDose(bySyncIdentifier: syncIdentifier) { (error) in
+            if let error = error {
+                completion(DoseStoreError.persistenceError(description: error, recoverySuggestion: nil))
+            } else {
+                NotificationCenter.default.post(name: DoseStore.valuesDidChange, object: self)
+                completion(nil)
```

Comment:
[nit] in every other case the `completion(...)` is called before the `NotificationCenter.default.post(...)` ... in reality the different timing may not matter but consider being consistent ...?